emscripten.py

```diff
@@ -1686,10 +1685,17 @@ def create_receiving(function_table_data, function_tables_defs, exported_impleme
     runtime_assertions = RUNTIME_ASSERTIONS
     # assert on the runtime being in a valid state when calling into compiled code. The only exceptions are
     # some support code
-    receiving = [f for f in exported_implemented_functions if f not in ('_memcpy', '_memset', '_emscripten_replace_memory', '__start_module')]
-    receiving = '\n'.join('var real_' + s + ' = asm["' + s + '"]; asm["' + s + '''"] = function() {''' + runtime_assertions + '''  return real_''' + s + '''.apply(null, arguments);
+    receiving_functions = [f for f in exported_implemented_functions if f not in ('_memcpy', '_memset', '_emscripten_replace_memory', '__start_module')]
+
+    wrappers = []
+    for name in receiving_functions:
+      wrappers.append('''\
+var real_%(name)s = asm["%(name)s"];
+asm["%(name)s"] = function() {%(runtime_assertions)s
+  return real_%(name)s.apply(null, arguments);
 };
-''' for s in receiving)
+''' % {'name': name, 'runtime_assertions': runtime_assertions})
```

Comment:
I didn't know `%` could take a dictionary

I prefer `'{name}({args})'.format({'name': name, 'args': args})` for this sort of thing, but it's probably moot. In particular it swaps paren ambiguity for curly ambiguity, so lgtm or whatever (although `.format` escapes curly ambiguity with `{{`... anyway neither option is great so who cares)